security/auth.go

```diff
@@ -139,3 +139,25 @@ func UserAuthHook(insecureMode bool, tlsState *tls.ConnectionState) (
 		return nil
 	}, nil
 }
+
+// CheckRequestUser extracts user from the request and verify.
```

Comment:
Maybe call this `CheckLocalRequestUser`? I feel like we need to do more to discourage the use of this method as opposed to the ones that use TLS.